src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java

```diff
@@ -84,17 +84,20 @@ public static void processEnvironmentVariable(
             boolean rubyOpts) throws CommandLineException {
 
         String value = System.getenv(name);
-        if (value != null && value.length() != 0) {
-            List<String> args = Arrays.asList(value.split("\\s+"));
-
-            if (args.size() != 0) {
-                new CommandLineParser(
-                        args,
-                        commandLineOptions,
-                        false,
-                        rubyOpts,
-                        true // let it parse and fail
-                ).processArguments();
+        if (value != null) {
+            value = value.trim();
+            if (value.length() != 0) {
```

Comment:
This would be clearer as `!value.isEmpty()` but that was already in the original code.